src/http.cc

```diff
@@ -1556,8 +1567,6 @@ HttpStateData::processReplyBody()
 
         case COMPLETE_NONPERSISTENT_MSG:
             debugs(11, 5, "processReplyBody: COMPLETE_NONPERSISTENT_MSG from " << serverConnection);
-            if (flags.chunked && !lastChunk)
```

Comment:
What exactly replaces this detection of missing 0-chunk? I am not spotting it in the new code.

Reply:
Adaptation aside, we mark fully received chunked messages in HttpStateData::decodeAndWriteReplyBody(). Responses which are not completed explicitly are marked as incomplete by StoreEntry::completeTruncated().